### PR TITLE
New feature: network policies (1.9.0)

### PR DESCRIPTION
**Pull request by anthosz**
_Descriptive description goes here._

Tasks:

 - [x] I've made changes
 - [x] I've bumped chart's version in `Chart.yaml`
 - [x] I've added changes to charts `CHANGELOG.md`
 - [x] I've run [`helm-docs`](https://github.com/norwoodj/helm-docs)

Fixes: #50

### CODE REVIEW

**Comment by MaikuMori** [comment]
It's currently failing test:

> Error: UPGRADE FAILED: template: gotenberg/templates/networkpolicy.yaml:1:14: executing "gotenberg/templates/networkpolicy.yaml" at <.Values.networkPolicy.enabled>: nil pointer evaluating interface {}.enabled

This happens because of how helm upgrades charts. To fix it you need to check if `networkPolicy` exists before checking if enabled in truthy.

This issue only happens on upgrades, not fresh installs.

**Comment by anthosz** [comment]
> It's currently failing test:
> 
> > Error: UPGRADE FAILED: template: gotenberg/templates/networkpolicy.yaml:1:14: executing "gotenberg/templates/networkpolicy.yaml" at <.Values.networkPolicy.enabled>: nil pointer evaluating interface {}.enabled
> 
> This happens because of how helm upgrades charts. To fix it you need to check if `networkPolicy` exists before checking if enabled in truthy.
> 
> This issue only happens on upgrades, not fresh installs.

Patch pushed + aligned with master

File: charts/gotenberg/templates/networkpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{{- if and .Values.networkPolicy .Values.networkPolicy.enabled }}
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ include "varnish-cache.fullname" . }}
```
Comment (MaikuMori) [inline charts/gotenberg/templates/networkpolicy.yaml:5]: Copy-paste?

File: charts/gotenberg/templates/networkpolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  labels:
+    {{- include "varnish-cache.labels" . | nindent 4 }}
```
Comment (MaikuMori) [inline charts/gotenberg/templates/networkpolicy.yaml:7]: Same here.

File: charts/gotenberg/templates/networkpolicy.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+spec:
+  podSelector:
+    matchLabels:
+      {{- include "gotenberg.selectorLabels" . | nindent 6 }}
+  policyTypes:
+    - Ingress
+    - Egress
+  ingress:
+{{- if .Values.networkPolicy.allowIngress }}
```
Comment (MaikuMori) [inline charts/gotenberg/templates/networkpolicy.yaml:16]: This seems a bit weird way to structure it, but I think it's ok for now. We can always accept a PR to change it.

**Review by MaikuMori — CHANGES_REQUESTED** [review]
I'm going to add a couple changes too so it's bundled in the same release.

File: charts/gotenberg/templates/networkpolicy.yaml (hunk 3 above)
Comment (anthosz) [reply charts/gotenberg/templates/networkpolicy.yaml:16]: Indeed, I copy paste from my PR for varnish, https://github.com/varnish/helm-varnish-cache/pull/24

It's not really clean but it allow to unblock when we use strict mode on cni :)

TODO: add a default schema that took into account the port & probably a matchexpressions based on ns or pod selector

Thank you!

**Pull request merged**